src/t8_cmesh/t8_cmesh_partition.cxx

```diff
@@ -131,7 +131,7 @@ t8_cmesh_gather_trees_per_eclass (const t8_cmesh_t cmesh, sc_MPI_Comm comm)
   t8_gloidx_t temp_trees_per_eclass[T8_ECLASS_COUNT];
   int ieclass;
 
-  T8_ASSERT (t8_cmesh_comm_is_valid (cmesh, comm));
+  T8_ASSERT (t8_cmesh_get_mpicomm (cmesh) == comm);
```

Comment:
I would still use `t8_cmesh_comm_is_valid ` here because of the straighforward naming
```suggestion
  T8_ASSERT (t8_cmesh_comm_is_valid (cmesh, comm);
```